pkg/broker/ingress.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package broker
+
+import (
+	"context"
+
+	eventingv1alpha1 "github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/pkg/provisioners"
+	"go.uber.org/zap"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	allowAny             = "allow_any"
+	allowRegisteredTypes = "allow_registered"
+)
+
+type IngressPolicy interface {
+	AllowMessage(namespace string, message *provisioners.Message) bool
+}
+
+type AllowAnyPolicy struct{}
+
+type AllowRegisteredPolicy struct {
+	logger *zap.SugaredLogger
+	client client.Client
+}
+
+func NewIngressPolicy(logger *zap.SugaredLogger, client client.Client, policy string) IngressPolicy {
+	return newIngressPolicy(logger, client, policy)
+}
+
+func newIngressPolicy(logger *zap.SugaredLogger, client client.Client, policy string) IngressPolicy {
+	switch policy {
+	case allowRegisteredTypes:
+		return &AllowRegisteredPolicy{
+			logger: logger,
+			client: client,
+		}
+	case allowAny:
+		return &AllowAnyPolicy{}
+	default:
+		return &AllowAnyPolicy{}
+	}
+}
+
+func (policy *AllowAnyPolicy) AllowMessage(namespace string, message *provisioners.Message) bool {
+	return true
+}
+
+func (policy *AllowRegisteredPolicy) AllowMessage(namespace string, message *provisioners.Message) bool {
+	eventType := &eventingv1alpha1.EventType{}
+	name := message.Headers["Ce-Eventtype"]
```

Comment:
This should probably be a constant.

According to https://github.com/cloudevents/spec/blob/master/http-transport-binding.md, the header name in the HTTP transport is `ce-type`. Is this transformed to `Ce-Eventtype` somewhere else, perhaps in the CloudEvents SDK?

Reply:
Yes, I agree. Should be some constant probably.

I asked Scott, he said that ce-type is spec version 0.2. For 0.1 is called Ce-Eventtype.

Reply:
Actually, Scott suggested we should use the SDK to decode the message. 
I'll do those changes... We need to do it as well for the filtering in triggers...